src/app/cli/src/init/mina_run.ml

```diff
@@ -445,27 +446,36 @@ let setup_local_server ?(client_trustlist = []) ?rest_server_port
                      ~f_witness:Transaction_witness.read_all_proofs_from_disk )
             in
             [%log trace]
-              ~metadata:[ ("work_spec", Snark_worker.Work.Spec.to_yojson work) ]
+              ~metadata:
+                [ ("work_spec", Work.Selector.Spec.Stable.Latest.to_yojson work)
+                ]
               "responding to a Get_work request with some new work" ;
             Mina_metrics.(Counter.inc_one Snark_work.snark_work_assigned_rpc) ;
             (work, key)) )
-    ; implement Snark_worker.Rpcs_versioned.Submit_work.Latest.rpc
-        (fun () (work : Snark_worker.Work.Result.t) ->
+    ; implement Snark_worker.Rpcs.Submit_work.Stable.Latest.rpc
+        (fun () (result : Work.Selector.Result.Stable.Latest.t) ->
           [%log trace] "received completed work from a snark worker"
             ~metadata:
-              [ ("work_spec", Snark_worker.Work.Spec.to_yojson work.spec) ] ;
-          log_snark_work_metrics work ;
-          Deferred.return @@ Mina_lib.add_work mina work )
-    ; implement Snark_worker.Rpcs_versioned.Failed_to_generate_snark.Latest.rpc
+              [ ( "work_spec"
+                , Work.Selector.Spec.Stable.Latest.to_yojson result.spec )
+              ] ;
+          log_snark_work_metrics result ;
+          (* TODO: plug in the correct cache DB here *)
```

Comment:
Here's one TODO left in this PR. But should be easy to resolve.

Reply:
I will open a follow-up issue to keep track of it.